### ISSUE TITLE
Leaflet with many tilelayers crashes Safari, Chrome and Firefox on iOS

### ISSUE THREAD

**Issue by jblarsen** [Leaflet/Leaflet#5749]
When many layers are added to a map WebKit based browsers crash when e.g. zooming. See:

http://playground-leaflet.rhcloud.com/janiz/1

using Leaflet 1.2.0.

The behavior is present on iPad, iPad Pro and iPhone although it is harder to provoke on iPhone. The issue disappears when hardware accelerated 3D transforms are disabled (L_DISABLE_3D=true). The problem seems to occur on the large iPad Pro before it happens on the smaller devices.

Although the playground example is contrived we do also have this issue on a real website.

I do realize that the issue is an iOS bug (or limitation) but Leaflet may want to work around it until fixed upstream.

**Comment by IvanSanchez** [comment]
Not sure how to approach this. How many tilelayers do you need to start triggering this problem?

**Comment by jblarsen** [comment]
I just checked. To consistently provoke the issue I need ~10 layers on an iPad Pro 12.9 inches with the 2048x1680 map width and height in the example. The iPad has a 2732-by-2048 Retina Display. Here is an updated example:

http://playground-leaflet.rhcloud.com/jimo/3

But it will also fail occasionally with less layers.

I don't know if it is this Safari issue:

https://medium.com/@MerciMichel/don-t-abuse-enabling-gpu-acceleration-on-ios-95520da2428

https://stackoverflow.com/questions/11831429/mobile-safari-on-ios-crashes-on-big-pages/13623612#13623612

I will try to figure out how to file a bug report to WebKit or Apple if the issue has not already been reported.

**Comment by perliedman** [comment]
This sounds similar to #4730 - might this be a duplicate, or any clear differences?

**Comment by jblarsen** [comment]
It may be the same issue. It is however only present when L_DISABLE_3D=false. But when 3D acceleration is disabled Leaflet will automatically disable e.g. zoom animations. I have not looked into the code but this probably means that only half the number of images are present in memory at the same time.

I can make a playground example with > double the number of layers needed to crash Safari when L_DISABLE_3D=false and check whether it then crashes with L_DISABLE_3D=true. If it does then it is probably the same issue?

**Comment by jblarsen** [comment]
Hi again. I tested with L_DISABLE_3D=true and more layers. It still crashes on an iPad Pro. This probably means that this issue is a duplicate of issue #4730. This issue should probably be closed then.

I practice this means that we will set L_DISABLE_3D=true for iOS devices. And we will try to limit the number of images (e.g. by making larger tiles or merging some of our transparent layers).

I have also filed a bug report at Apple but haven't heard anything from them yet.
